plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/util/CodeWhispererUtil.kt

```diff
@@ -347,3 +354,88 @@ object CodeWhispererUtil {
 enum class CaretMovement {
     NO_CHANGE, MOVE_FORWARD, MOVE_BACKWARD
 }
+
+fun getDiagnosticsType(message: String): String {
+    val lowercaseMessage = message.lowercase()
+
+    val diagnosticPatterns = mapOf(
+        "TYPE_ERROR" to listOf("type", "cast"),
+        "SYNTAX_ERROR" to listOf("expected", "indent", "syntax"),
+        "REFERENCE_ERROR" to listOf("undefined", "not defined", "undeclared", "reference", "symbol"),
+        "BEST_PRACTICE" to listOf("deprecated", "unused", "uninitialized", "not initialized"),
+        "SECURITY" to listOf("security", "vulnerability")
+    )
+
+    return diagnosticPatterns
+        .entries
+        .firstOrNull { (_, keywords) ->
+            keywords.any { lowercaseMessage.contains(it) }
+        }
+        ?.key ?: "OTHER"
+}
+
+fun convertSeverity(severity: HighlightSeverity): String = when {
+    severity == HighlightSeverity.ERROR -> "ERROR"
+    severity == HighlightSeverity.WARNING ||
+        severity == HighlightSeverity.WEAK_WARNING -> "WARNING"
+    severity == HighlightSeverity.INFORMATION -> "INFORMATION"
+    severity.toString().contains("TEXT", ignoreCase = true) -> "HINT"
```

Comment:
```suggestion
    severity.name == "TEXT" -> "HINT"
```
?

Reply:
I think this changed the logic right? It was trying to look contains not exact match

Reply:
my question is if we can do equality or does it need to be a contains check

Reply:
<img width="767" alt="Screenshot 2025-07-07 at 2 30 33 PM" src="https://github.com/user-attachments/assets/2e637be2-83b6-4510-95f1-c2185c228c79" />
I've run a local test to check what's the actual type, yes I can use the equality here